pentaho-osgi-utils-impl/src/main/java/org/pentaho/osgi/impl/KarafBlueprintWatcherImpl.java

```diff
@@ -74,8 +75,12 @@ public KarafBlueprintWatcherImpl( BundleContext bundleContext ) {
           }
           if ( unloadedBlueprints.size() > 0 ) {
             if ( System.currentTimeMillis() - timeout > entryTime ) {
-              throw new IKarafBlueprintWatcher.BlueprintWatcherException(
-                  "Timed out waiting for blueprints to load: " + StringUtils.join( unloadedBlueprints, "," ) );
+              if ( BarrierBeanProcessor.getInstance().isAvailable( "KarafFeatureWatcherBarrier" ) ) {
```

Comment:
yea din't like this isAvailable. I don't wait anyone calling BarrierBeanProcessor. They should be interactive with the Barrier Manager